tools/osbuilder/rootfs-builder/centos/config.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
+# Copyright (c) 2018 Intel Corporation, 2021 IBM Corp.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+OS_NAME=centos
+OS_VERSION=${OS_VERSION:-stream9}
+PACKAGES=chrony
+[ "$AGENT_INIT" = no ] && PACKAGES+=" systemd"
+[ "$SECCOMP" = yes ] && PACKAGES+=" libseccomp"
+
+# Container registry tag is different from metalink repo, e.g. "stream9" => "9-stream"
+os_repo_version="$(sed -E "s/(stream)(.+)/\2-\1/" <<< "$OS_VERSION")"
```

Comment:
The `os_repo_version` is actually not needed. Instead you can use the `$stream` variable of `DNF`.

I just checked it evaluates to `9-stream` in the CentOS Stream 9 image: 
```
$ podman run --rm -it quay.io/centos/centos:stream9 cat /etc/dnf/vars/stream
9-stream

```
So the metalink can be rewrite as `https://mirrors.centos.org/metalink?repo=centos-baseos-\$stream&arch=\$basearch`

To be honest I don't understand why we still need to create a custom dnf configure file. I saw that `generate_dnf_config()` it was introduced in 2017 in commit 5b8478c4c as a mean of allow users to overwrite the default yum repository. I wonder if such as use case is still valid and has users....